pkg/collector/corechecks/snmp/internal/profile/profile_yaml.go

```diff
@@ -30,21 +31,20 @@ const profilesJSONGzipFile = "profiles.json.gz"
 
 var defaultProfilesMu = &sync.Mutex{}
 
-var globalProfileConfigMap profileConfigMap
-
 // loadYamlProfiles will load the profiles from disk only once and store it
 // in globalProfileConfigMap. The subsequent call to it will return profiles stored in
 // globalProfileConfigMap. The mutex will help loading once when `loadYamlProfiles`
 // is called by multiple check instances.
-func loadYamlProfiles() (profileConfigMap, error) {
+func loadYamlProfiles() (ProfileConfigMap, error) {
 	defaultProfilesMu.Lock()
 	defer defaultProfilesMu.Unlock()
 
-	if globalProfileConfigMap != nil {
-		log.Debugf("loader default profiles from cache")
-		return globalProfileConfigMap, nil
+	profileConfigMap := GetGlobalProfileConfigMap()
```

Comment:
would this include both `default` and `user` yaml profiles ?

Reply:
Yes, it does.

Few lines below you have the `SetGlobalProfileConfigMap(profiles)`, the profiles here are from ` getDefaultProfilesDefinitionFiles()` that load/resolve both default and user profiles